src/amoc_api_scenario_handler.erl

```diff
@@ -10,112 +10,125 @@
+            case ContainUsers of
+                true ->
+                    Users = proplists:get_value(<<"users">>, JSON),
+                    State2 = State#state{users = Users},                    
+                    {false, Req2, State2};
+                false ->
+                    {true, Req2, State}
+            end
+    end;
+
+malformed_request(_, Req, State) ->
+    {false, Req, State}.
+
+-spec resource_exists(cowboy:req(), state()) ->
+    {boolean(), cowboy:req(), state()}.
+resource_exists(Req, State = #state{resource = Resource}) ->
+    {ok, Files} = file:list_dir("scenarios/"),
+    Pred = fun (File) -> File == Resource ++ ".erl" end,
+    case lists:filter(Pred, Files) of
+        [] ->
+            {false, Req, State};
+        [_File] ->
+            {true, Req, State}
     end.
 
-%%%%%%%%%%%%%%
-% Request processing functions
-%%%%%%%%%%%%%%
--spec process_json(jsx:json_term(), state()) -> {integer(), jsx:json_text()}.
-process_json(Term, #state{action = start}) ->
-   ScenarioB = proplists:get_value(<<"scenario">>, Term),
-    Users = proplists:get_value(<<"users">>, Term),
-    
-    Scenario = erlang:binary_to_atom(ScenarioB, utf8),
-    
-    case code:load_file(Scenario) of
-        {error, nofile} ->
-            {500, jsx:encode([{<<"error">>, <<"module_not_exists">>}])};
-        {module, Scenario} -> 
-            _ = amoc_dist:do(Scenario, 1, Users),
-            {200, jsx:encode([{ScenarioB, <<"ok">>}])}
-    end;
 
-process_json(_Term, #state{action = stop}) ->
-    {200, jsx:encode("ok")};
-
-process_json(Term, #state{action = load}) ->
-    ScenarioB = proplists:get_value(<<"scenario">>, Term),
-    ModuleB = proplists:get_value(<<"module_source">>, Term),
-    Scenario = erlang:binary_to_atom(ScenarioB, utf8),
-    ScenarioPath = "scenarios/" ++ erlang:atom_to_list(Scenario) ++ ".erl",
-    ok = file:write_file(ScenarioPath, ModuleB, [write]),
-    case compile:file(ScenarioPath, [{outdir, "ebin"}]) of
-        {ok, Scenario} ->
-            code:purge(Scenario),
-            {200, jsx:encode([{ScenarioB, <<"loaded">>}])};
-        error ->
-            {500, jsx:encode([{<<"error">>, <<"compilation_error">>}])}
-    end;
 
-process_json(_Term, #state{action = ping_nodes}) ->
-    Nodes = amoc_dist:ping_nodes(),
-    {200, jsx:encode([{<<"nodes">>, Nodes}])};
+%% Request processing functions
+
+-spec handle_get(cowboy:req(), state()) -> 
+    {string() | halt, cowboy:req(), state()}.
+handle_get(Req0, State = #state{resource = Resource}) ->
+    %% Need to retrive state of module here
+    Reply = jsx:encode([{module, Resource}]),
+    {Reply, Req0, State}.
+
+
+-spec handle_patch(cowboy:req(), state()) ->
+    {string() | halt | ok, cowboy:req(), state()}.
+handle_patch(Req0, State = #state{resource = Resource, users = Users}) ->
+    Scenario = erlang:list_to_atom(Resource),
+    code:purge(Scenario),
+    Result = code:load_file(Scenario),
```

Comment:
I don't understand where the file is actually saved. The code:load_file/1 will try to load an existing module from a file with the same name. It will look for the file in the code search paths. 

Have you tested that it actually works?